api/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,130 @@
+	// +kubebuilder:validation:Required
+	// Introduce an EnhancedNodeSelector that supports combining label-based
+	// matching with resource field conditions.
+	// For example:
+	// enhancedNodeSelector:
+	//  # 1. Label selector (compatible with existing metav1.LabelSelector)
+	//  labelSelector:
+	//    matchLabels:
+	//      nvidia.com/gpu.family: ada-lovelace
+	//    matchExpressions:
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      memory: ">=200Gi"
+	//      nvidia.com/gpu: ">=1"
+	//      cpu: ">=32"
+	//    # Optional: also support capacity (if total capacity—not just allocatable—is desired)
+	//    # capacity:
+	//    #   memory: ">=256Gi"
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node.
+	// If two different counts are specified for the same (Node, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +kubebuilder:validation:MinItems=1
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirementSpec defines resource requirements for a node.
+type ResourceRequirementSpec struct {
+	// Allocatable defines the allocatable resources for a node.
+	// +kubebuilder:validation:Required
+	Allocatable map[string]string `json:"allocatable,omitempty"`
+	// Capacity defines the capacity resources for a node.
+	// +kubebuilder:validation:Optional
+	Capacity map[string]string `json:"capacity,omitempty"`
+}
+
+// EnhancedNodeSelector defines node selector with label selector and resource requirements.
+type EnhancedNodeSelector struct {
+	// LabelSelector defines the label selector for a node.
+	// +kubebuilder:validation:Required
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+	// ResourceRequirements defines the resource requirements for a node.
+	// +kubebuilder:validation:Optional
+	ResourceRequirements *ResourceRequirementSpec `json:"resourceRequirements,omitempty"`
+}
+
+type CountForLauncher struct {
+	// LauncherConfigName references the name of the LauncherConfig this policy applies to.
+	// +kubebuilder:validation:Required
+	LauncherConfigName string `json:"launcherConfigName"`
+
+	// LauncherCount is the total number of launcher pods to maintain.
+	// +kubebuilder:validation:Required
+	LauncherCount int32 `json:"launcherCount"`
+}
+
+type LauncherPoolPolicyStatus struct {
+	// `observedGeneration` is the `metadata.generation` last seen by the controller.
+	// +optional
+	ObservedGeneration int32 `json:"observedGeneration,omitempty"`
```

Comment:
Oops, I am sorry, I goofed up in an earlier comment recommending this. The type should actually be `int64`. See https://github.com/kubernetes/apimachinery/blob/v0.34.3/pkg/apis/meta/v1/types.go#L177